testlab1-2.c

```diff
@@ -0,0 +1,304 @@
+    {"\xE0\xE1 \xE2\xE3\n\xE0\xE1 \xE2\xE3\xE0\xE1\xE2\xE3", 7, {0, 0, 0, 0, 0, 1, 5}}, // аб вг\nаб вгабвг
+    {"\xE0\xE1 \xE2\xE3\n\xE0\xE1\xE2\xE3\n\xE0\xE1 \xE2\xE3", 9, {0, 0, 0, 0, 0, 1, 2, 6, 5}}, // аб вг\nабвг\nаб вг
+    {"000001\n000000000010", 18,{0, 1, 2, 3, 4, 0, 1, 5, 2, 5, 3, 5, 4, 5, 5, 5, 6, 6}}
+};
+
+static int FeedFromArray(void)
+{
+    FILE *const in = fopen("in.txt", "w+");
+    if (!in) {
+        printf("can't create in.txt. No space on disk?\n");
+        return -1;
+    }
+    fprintf(in, "%s", testInOut[testN].in);
+    fclose(in);
+    return 0;
+}
+
+static int CheckFromArray(void)
+{
+    FILE *const out = fopen("out.txt", "r");
+    int i, passed = 1;
+    if (!out) {
+        printf("can't open out.txt\n");
+        testN++;
+        return -1;
+    }
+    for (i = 0; i < testInOut[testN].n; i++) {
+        int n;
+        if (ScanInt(out, &n) != Pass || testInOut[testN].out[i] != n) {
+            passed = 0;
+            printf("wrong output -- ");
+            break;
+        }
+    }
+    if (passed) {
+        passed = !HaveGarbageAtTheEnd(out);
+    }
+    fclose(out);
+    if (passed) {
+        printf("PASSED\n");
+        testN++;
+        return 0;
+    } else {
+        printf("FAILED\n");
+        testN++;
+        return 1;
+    }
+}
+
+static int FeederBigRand1(void)
+{
+    FILE *const in = fopen("in.txt", "w");
+    DWORD t;
+    unsigned int bigTestN = testN - sizeof(testInOut) / sizeof(testInOut[0]);
+    char pattern[] = "0123456789abcdef";
+    if (!in) {
+        printf("can't create in.txt. No space on disk?\n");
+        return -1;
+    }
+    printf("Creating large text with seed %u... ", seed[bigTestN]);
+    fflush(stdout);
+    srand(seed[bigTestN]);
+    t = GetTickCount();
+    if(fputs("0123456789abcdef\n", in) == EOF) {
+        printf("can't write in in.txt. No space on disk?\n");
+        return -1;
+    }
+    unsigned int r;
+    for(unsigned int i = 1; i < 1024*1024*8; i++) {
+        r = (unsigned int)rand()%17;
+        if(r == 0) {
+            if(putc(' ', in) == EOF) {
+                printf("can't write in in.txt. No space on disk?\n");
+                return -1;
+            }
+        }else {
+            if (fwrite(pattern, 1, r, in) != r) {
+                printf("can't write in in.txt. No space on disk?\n");
+                return -1;
+            }
+        }
+
+    }
+    if(fputs("0123456789abcdef", in) == EOF) {
+        printf("can't write in in.txt. No space on disk?\n");
+        return -1;
+    }
+    fclose(in);
+    t = RoundUptoThousand(GetTickCount() - t);
+    printf("done in T=%u seconds. Starting exe with timeout 10*T... ", (unsigned)t/1000);
+    LabTimeout = (int)t*10;
+    fflush(stdout);
+    return 0;
+}
+
+static int CheckerBigRand1(void)
+{
+    FILE *const out = fopen("out.txt", "r");
+    unsigned int passed = 1;
+    unsigned int bigTestN = testN - sizeof(testInOut) / sizeof(testInOut[0]);
```

Comment:
if bigTestN < 0 or bigTestN >= size of seeds, then print a message about internal error, and quit